src/plone/restapi/services/content/add.py

```diff
@@ -74,6 +77,11 @@ def reply(self):
         if temporarily_wrapped:
             obj = aq_base(obj)
 
+        # ManagePortal is required to set the uid of an object during creation
+        sm = getSecurityManager()
+        if uid and sm.checkPermission(ManagePortal, self.context):
```

Comment:
@ericof I am wondering if we should fail silently here or issue a message (at least on the console) or even fail hard when someone tries to set a UID without the proper permission?

Reply:
Please fail with an `Unauthorized` and a reason.

Reply:
👍🏼